packages/appium/lib/schema/schema.js

```diff
@@ -4,12 +4,42 @@ import Ajv from 'ajv';
 import addFormats from 'ajv-formats';
 import _ from 'lodash';
 import path from 'path';
-import { DRIVER_TYPE, PLUGIN_TYPE } from '../extension-config';
-import { ReadonlyMap } from '../utils';
+import { DRIVER_TYPE, PLUGIN_TYPE } from '../constants';
 import appiumConfigSchema from './appium-config-schema';
 import { APPIUM_CONFIG_SCHEMA_ID, ArgSpec, SERVER_PROP_NAME } from './arg-spec';
 import { keywords } from './keywords';
 
+/**
+ * Key/value pairs go in... but they don't come out.
+ *
+ * @template K,V
+ * @extends {Map<K,V>}
+ */
+export class RoachHotelMap extends Map {
```

Comment:
I renamed `ReadonlyMap` to `RoachHotelMap` because @mykola-mokhnach pointed out that it was a misnomer.  It's _not_ "read-only", since you can write to it.  But you _can't_ take things out or change what's already there.